rllib/connectors/tests/test_agent.py

```diff
@@ -381,44 +330,133 @@ def test_vr_connector_with_multiple_buffers(self):
-                    check(for_action["context_act"], act_arrs[i - context_len : i])
+            else:
+                obs_list.pop(0)
+                act_list.pop(0)
+                obs_list.append(obs_arrs[t])
+                act_list.append(act_arrs[t - 1])
+
+            self.assertTrue("context_next_obs" not in for_action)
+            check(for_action["context_obs"], np.stack(obs_list)[None])
+            check(for_action["context_act"], np.stack(act_list[:-1])[None])
+
+    @pytest.mark.skip("test is not ready yet.")
+    def test_connector_pipline_with_view_requirement(self):
+        config = (
+            PPOConfig()
+            .environment(env="CartPole-v0")
+            .training(
+                model=dict(
+                    use_lstm=True,
+                    lstm_use_prev_action=True,
+                    lstm_use_prev_reward=True,
+                ),
+            )
+            .rollouts(create_env_on_local_worker=True)
+        )
+        algo = PPO(config)
+        rollout_worker = algo.workers.local_worker()
+        policy = rollout_worker.get_policy()
+        env = rollout_worker.env
+
+        # create a connector context
+        ctx = ConnectorContext(
+            view_requirements=policy.view_requirements,
+            config=policy.config,
+            initial_states=policy.get_initial_state(),
+            is_policy_recurrent=policy.is_recurrent(),
+            observation_space=policy.observation_space,
+            action_space=policy.action_space,
+        )
+
+        # build chain of connectors
+        connectors = [
+            ObsPreprocessorConnector(ctx),
+            ClipRewardAgentConnector(ctx, False, 1.0),
+            FlattenDataAgentConnector(ctx),
+            StateBufferConnector(ctx),
+            ViewRequirementAgentConnector(ctx),
+        ]
+        pipeline = AgentConnectorPipeline(ctx, connectors)
+
+        name, params = pipeline.to_config()
+        restored = get_connector(ctx, name, params)
+        self.assertTrue(isinstance(restored, AgentConnectorPipeline))
+        for cidx, c in enumerate(connectors):
+            check(restored.connectors[cidx].to_config(), c.to_config())
+
+        # simulate a rollout
+        n_steps = 10
+        obs = env.reset()
+        env_out = AgentConnectorDataType(
+            0, 1, {SampleBatch.NEXT_OBS: obs, SampleBatch.T: -1}
+        )
+        agent_obs = pipeline([obs])
+        t = 0
+        while t < n_steps:
+            eval_batch = agent_obs.data.for_action
+            action, state, extra_fetch = policy.compute_actions_from_input_dict(
+                eval_batch
+            )
+            next_obs, rewards, dones, info = env.step(action)
+            env_out_dict = {
+                SampleBatch.NEXT_OBS: next_obs,
+                SampleBatch.REWARDS: rewards,
+                SampleBatch.DONES: dones,
+                SampleBatch.INFO: info,
+                SampleBatch.ACTIONS: action,
```

Comment:
ah we are providing ACTIONS column here.
so state_buffer connector is not actually doing anything in this setup, since it's not receiving any policy output callbacks.
it's fine. may be something we will come back and improve later.